core/config/project_settings.cpp

```diff
@@ -522,9 +521,7 @@ Error ProjectSettings::_setup(const String &p_path, const String &p_main_pack, b
 		// OS will call ProjectSettings->get_resource_path which will be empty if not overridden!
 		// If the OS would rather use a specific location, then it will not be empty.
 		resource_path = OS::get_singleton()->get_resource_dir().replace("\\", "/");
-		if (!resource_path.is_empty() && resource_path[resource_path.length() - 1] == '/') {
-			resource_path = resource_path.substr(0, resource_path.length() - 1); // Chop end.
-		}
+		resource_path = resource_path.trim_suffix("/");
```

Comment:
Old code says that if the resource path isn't empty and the last character is '/', then we should remove that last character from the string. `trim_suffix()` can do the same.